libs/core/src/main/java/org/opensearch/core/common/io/stream/StreamOutput.java

```diff
@@ -804,6 +804,23 @@ private static Class<?> getGenericType(Object value) {
         }
     }
 
+    /**
+     * Returns the registered writer for the given class type.
+     */
+    @SuppressWarnings("unchecked")
+    public static <W extends Writer<?>> W getGenericValueWriterByClass(Class<?> type) {
```

Comment:
Better to follow the convention we have (fe `WriteableRegistry::getWriter(type)`);
```suggestion
    public static <W extends Writer<?>> W getWriter(Class<?> type) {
```

Reply:
My intent was to keep 'generic' in the name as other non-generic writers (eg. well-defined methods like `writeLong`, `writeDouble`, etc.) differ from generic writers in the sense that they don't write the 'ordinal' byte representing the type of the value.

Let me know if you still prefer `getWriter` instead. In favor of your point though, eventually all writers should move to the `WriteableRegistry` class.

Reply:
Hmm.. I'm the one that originally asked for this to be renamed to `getWriterByClass`..  I was originally thinking of avoiding the method clash but it does make more sense for them to align since the `WriteableRegistry` is package private anyway.

+1.. sorry for the repeat refactor @ketanv3

Reply:
Updated!